### PR TITLE
[Assist] Remove the empty assist message

### PR DESCRIPTION
**Pull request by jakule**
Assist shows an empty message at the beginning of each conversation when reading it from DB. This PR fixes that behavior and adds a test to prevent this from happening in the future.

### CODE REVIEW

**Review by xacrimon — APPROVED** [review]

**Review by hugoShaka — APPROVED** [review]

File: lib/assist/assist.go [hunk 1]
```diff
@@ -59,6 +58,18 @@ const (
 	MessageKindError MessageType = "CHAT_MESSAGE_ERROR"
 )
 
+type PluginGetter interface {
+	PluginsClient() pluginsv1.PluginServiceClient
+}
+
+type AssistantService interface {
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/assist/assist.go]: Added

File: lib/assist/assist.go [hunk 2]
```diff
@@ -67,7 +78,7 @@ type Assist struct {
 }
 
 // NewAssist creates a new Assist client.
-func NewAssist(ctx context.Context, proxyClient auth.ClientI,
+func NewAssist(ctx context.Context, proxyClient PluginGetter,
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/assist/assist.go]: "Touch a line and the owner of it you become" :laughing: Changed

File: lib/assist/assist_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		// The first message is the welcome message.
+		_, err = chat.ProcessComplete(ctx, func(kind MessageType, payload []byte, createdTime time.Time) error {
+			require.Equal(t, MessageKindAssistantMessage, kind)
+			require.Contains(t, string(payload), "Hey, I'm Teleport")
+			return nil
+		}, "")
+		require.NoError(t, err)
+	})
+
+	t.Run("new conversation is not complete", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/assist/assist_test.go]: Changed

**Review by rosstimothy — APPROVED** [review]

File: lib/assist/assist.go [hunk 4]
```diff
@@ -59,15 +58,29 @@ const (
 	MessageKindError MessageType = "CHAT_MESSAGE_ERROR"
 )
 
+// PluginGetter is the minimal interface used by the chat to interact with the plugins service in the backend.
+type PluginGetter interface {
+	PluginsClient() pluginsv1.PluginServiceClient
+}
+
+// AssistantService is the minimal interface used by the chat to interact with the Assist service in the backend.
+type AssistantService interface {
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/assist/assist.go]: I changed the name, but this is not the same thing. This package is called `assist`, but the `AssistantService` doesn't refer to this package, but the `Assist` service that lives in the backend. I renamed it to `MessageService` as all receivers are related to messages.

**Pull request merged**